src/core/dev/qp_mgr_eth_mlx5.h

```diff
@@ -194,6 +194,8 @@ class qp_mgr_eth_mlx5 : public qp_mgr_eth {
         return NULL;
     }
 
+    virtual bool is_rq_empty() const { return (m_mlx5_qp.rq.head == m_mlx5_qp.rq.tail); }
```

Comment:
[question] Is it safe to check without protection?

Reply:
This method is called from safe place.